### PR TITLE
[fix](thirdparty) fix compile error: missing libbrpc.a

### LINKED ISSUE #11351: [Bug] Compiling doris error: missing libbrpc.a
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

Compiling doris error:
```
-- Configuring done
-- Generating done
-- Build files have been written to: /home/disk3/lzl/icode/baidu/bdg/doris/core/be/build_Debug
[0/2] Re-checking globbed directories...
ninja: error: '/home/disk3/lzl/icode/baidu/bdg/doris/core/thirdparty/installed/lib64/libbrpc.a', needed by 'src/service/doris_be', missing and no known rule to make it
```

### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #11351

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [x] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -345,7 +345,7 @@ build_protobuf() {
         ldflags="-L${TP_LIB_DIR} -static-libstdc++ -static-libgcc -Wl,--undefined=pthread_create"
     fi
 
-    CXXFLAGS="-O2 -I${TP_INCLUDE_DIR}" \
+    CXXFLAGS="-fPIC -O2 -I${TP_INCLUDE_DIR}" \
```
Comment (yiguolei) [inline thirdparty/build-thirdparty.sh:348]: why add -fPIC here? I find pxl do remove it few days ago.

**Comment by BiteTheDDDDt** [comment]
brpc will build both shared and static





------------------ Original ------------------
From: yiguolei ***@***.***&gt;
Date: Sat,Jul 30,2022 8:05 PM
To: apache/doris ***@***.***&gt;
Cc: Pxl ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [apache/doris] [fix] fix compile error: missing libbrpc.a (PR #11353)






@yiguolei commented on this pull request.


In thirdparty/build-thirdparty.sh:
&gt; @@ -345,7 +345,7 @@ build_protobuf() {         ldflags="-L${TP_LIB_DIR} -static-libstdc++ -static-libgcc -Wl,--undefined=pthread_create"     fi -    CXXFLAGS="-O2 -I${TP_INCLUDE_DIR}" \+    CXXFLAGS="-fPIC -O2 -I${TP_INCLUDE_DIR}" \
why add -fPIC here? I find pxl do remove it few days ago.

**Comment by adonis0147** [comment]
Some other issues also exist in the script. See #11357 .

**Comment by adonis0147** [comment]
Merge this one first.

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**